File: software/bmi_java/src/main/java/BMICalculator.java
// BMICalculator.java
public class BMICalculator {
    public static double calculateBMI(double weight, double height) {
        if (height < 0) {
            throw new IllegalArgumentException("Height must be greater than zero");
        }
        return weight / (height * height);
    }

    public static void main(String[] args) {
        java.util.Scanner scanner = new java.util.Scanner(System.in);
        System.out.print("Enter weight in kilograms: ");
        double weight = scanner.nextDouble();
        System.out.print("Enter height in meters: ");
        double height = scanner.nextDouble();
        scanner.close();

        try {
            double bmi = calculateBMI(weight, height);
            System.out.printf("Your BMI is: %.2f%n", bmi);
        } catch (IllegalArgumentException e) {
            System.err.println("Error: " + e.getMessage());
        }
    }
}

File: software/bmi_cpp/bmi_calculator.cpp
// bmi_calculator.cpp
#include <iostream>
#include <stdexcept>

double calculate_bmi(double weight, double height) {
    if (height < 0) {
        throw std::invalid_argument("Height must be greater than zero");
    }
    return weight / (height * height);
}

#ifndef CATCH_CONFIG_MAIN
int main(int argc, char * argv[]) {
    double weight, height;
    std::cout << "Enter weight in kilograms: ";
    std::cin >> weight;
    std::cout << "Enter height in meters: ";
    std::cin >> height;

    try {
        double bmi = calculate_bmi(weight, height);
        std::cout << "Your BMI is: " << bmi << std::endl;
    } catch (const std::invalid_argument& e) {
        std::cerr << "Error: " << e.what() << std::endl;
    }

    return 0;
}
#endif